code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -59,6 +59,9 @@
 
 	populate_contents()
 
+	for(var/obj/item/L in return_inv()) //marks all items in storage as being such
+		L.in_storage = TRUE
+
```

Comment:
This *might* be better on item init, just so we're not performing this on empty storages. The item would check it's location and see if it's in storage or not, and set the var accordingly.